### ISSUE TITLE
Cannot use global mutate with useSWRInfinite

### ISSUE THREAD

**Issue by nicholaschiang** [vercel/swr#1156]
# Bug report

## Description / Observed Behavior

There is no way to mutate data fetched with `useSWRInfinite` without first fetching the data client-side. I saw [this answer](https://github.com/vercel/swr/pull/435#issuecomment-658219974) by @shuding and it definitely works, but it triggers a fetch when all I want to do is mutate data (if it has already been fetched).

## Expected Behavior

Instead, I should be able to mutate local data without triggering a fetch. The global `mutate` function should allow me to mutate data **that has already been fetched** (e.g. on a page that uses `useSWRInfinite`) on a page that should not trigger a fetch of that data.

My (pretty common I think) use-case:
1. I have a list of emails fetched using `useSWRInfinite` on a `/feed` page.
2. When the user opens an email, they can modify it on a `/messages/[id]` page.
3. I want to mutate the data shown on the `/feed` page when the data is edited on the `/messages/[id]` page.

But, the user can also open the `/messages/[id]` page directly, in which case the `/feed` page will never have been opened and `useSWRInfinite` will never have been called. In this case, I don't want to trigger a fetch with `useSWRInfinite`.

## Repro Steps / Code Example

This is the project I'm working on: [`nicholaschiang/hammock`](https://github.com/nicholaschiang/hammock).

## Additional Context

SWR version: 0.5.5
Related to #435.

**Comment by alizahid** [comment]
I'm using SWR on my Next.js and React Native apps, and while I might be able to pass `mutate` between components in Next.js, I can't do the same on React Native, because my mutator is on another screen, and React Navigation doesn't allow functions to pass between screens.

What's the workaround for this? Without having to reload everything every time?

**Comment by sjchmiela** [comment]
I was able to find a workaround for this today, see [comment](https://github.com/vercel/swr/issues/662#issuecomment-845263259):

tl;dr: use this specific `infiniteKey` formula:

```ts
import { mutate, cache } from "swr";

const KEY_FOR_FIRST_PAGE = ...;
const [serializedFirstPageKey] = cache.serialize(KEY_FOR_FIRST_PAGE);
const infiniteKey = ["inf", serializedFirstPageKey];

mutate(infiniteKey, data => { ... }, false);
```

**Comment by koba04** [comment]
A solution in https://github.com/vercel/swr/issues/1156#issuecomment-845264050 might work at this time, but it would be broken in the future. For example, https://github.com/vercel/swr/pull/1174 breaks your solution.
I think SWR should provide a solution to generate the key, which doesn't depend on the internal implementation if there is any valid use-case that the global `mutate` has to use.

**Comment by shuding** [comment]
I'm not sure if it's a good idea to expose internal implementation details (things like `inf`), as they will likely to change over time. The idea of #1174 is to support use cases like `mutate(regex)` or `mutate(filterFn)` in the future. 

While `useSWRInfinite` is a truly special case, I'd propose a `mutateInfinite` method so you can do `mutateInfinite(KEY_FOR_FIRST_PAGE)` or `mutateInfinite(getKey)`. It basically hides the `inf` inside.

**Comment by Develliot** [comment]
This would also solve my scenario mentioned here https://github.com/vercel/swr/discussions/631

**Comment by jamneck** [comment]
I see this is closed, but it doesn't seem to be part of a release or documented.

**Comment by shuding** [comment]
There is a (unstable) helper in 1.x versions:

```js
import { unstable_serialize } from 'swr/infinite'

mutate(unstable_serialize(getKey))
```

We still need to finalize the API design though.

**Comment by cgxxv** [comment]
> 

This works for me in 2.0.0 version, so I think it's maybe a bug, so do you guys have a better solution to fix this.

**Comment by spacepolice10** [comment]
Any plans on providing filter function for useSWRInfinite?
